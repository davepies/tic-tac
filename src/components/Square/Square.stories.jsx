import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import FancySquare from '.';

storiesOf('FancySquare', module)
  .add('Front', () => <FancySquare onClick={action('clicked the square')} value='X' />)
