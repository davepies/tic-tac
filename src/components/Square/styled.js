import styled from 'styled-components';

export const Button = styled.button`
    width: ${({ theme }) => theme.dimensions.squareSize};
    line-height: ${({ theme }) => theme.dimensions.squareSize};
    height: ${({ theme }) => theme.dimensions.squareSize};

    font-size: ${({ theme }) => theme.dimensions.squareFontSize};
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;

    background: ${({ theme }) => theme.colors.primary};
    color: ${({ theme }) => theme.colors.secondary};
    border: 1px solid ${({ theme }) => theme.colors.secondary};

    font-weight: bold;
    text-align: center;

    cursor: ${({ occupied }) => occupied ? 'no-drop' : 'pointer'};

    :focus {
        outline: none;
    }
`;