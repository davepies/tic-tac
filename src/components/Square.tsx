import React from 'react';
import styled from 'styled-components';

const SquareButton = styled.button<{ occupied?: boolean }>`
    background: #fff;
    border: 1px solid #999;
    float: left;
    font-size: 24px;
    font-weight: bold;
    line-height: 34px;
    height: 34px;
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;
    text-align: center;
    width: 34px;
    cursor: ${({ occupied }) => occupied ? 'no-drop' : 'pointer'};

    :focus {
        outline: none;
    }
`;

export interface ISquareProps {
    value: string | null;
    onClick?: () => void;
}

const Square = (props: ISquareProps): JSX.Element => {
    const { value, onClick } = props;
    const isOccupied = value !== null;
    return (
        <SquareButton onClick={onClick} occupied={isOccupied}>
            {value}
        </SquareButton>
    );
}

export default Square;
