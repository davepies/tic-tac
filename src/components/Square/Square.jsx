import React from 'react';

import { Button } from './styled';

const Square = (props) => {
    const { value, onClick } = props;
    const isOccupied = value !== null;
    return (
        <Button onClick={onClick} occupied={isOccupied}>
            {value}
        </Button>
    );
}

export default Square;
