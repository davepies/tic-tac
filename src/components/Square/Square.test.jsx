import React from 'react';
import {shallow} from 'enzyme';
import sinon from 'sinon';

import Square from './';

test('<Square /> renders', () => {
    const wrapper = shallow(<Square value={'X'} />);

    expect(wrapper).toMatchSnapshot();
});

test('<Square /> contains passed value', () => {
    const value = 'XYZ';
    const wrapper = shallow(<Square value={value} />);

    expect(wrapper.contains(value)).toBe(true);
});

test('<Square /> calls onClick', () => {
    const value = 'XYZ';
    const onClick = sinon.spy();

    const wrapper = shallow(<Square value={value} onClick={onClick} />);
    wrapper.simulate('click');

    expect(onClick.callCount).toBe(1);
});