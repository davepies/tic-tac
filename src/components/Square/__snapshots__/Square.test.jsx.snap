// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> renders 1`] = `
<styled__Button
  occupied={true}
>
  X
</styled__Button>
`;
